menas/ui/service/EntityDialog.js

```diff
@@ -0,0 +1,352 @@
+    this.oDialog.open();
+
+    this.schemaService.getList(this.oDialog).then(() => {
+      let current = this.oController._model.getProperty("/currentDataset");
+
+      current.isEdit = true;
+      current.title = "Edit";
+      this.schemaService.getAllVersions(current.schemaName, this.oController.byId("schemaVersionSelect"));
+
+      this.oDialog.setModel(new sap.ui.model.json.JSONModel(jQuery.extend(true, {}, current)), "entity");
+      this.oDialog.setBusy(false);
+    });
+  }
+
+}
+
+class SchemaDialog extends EntityDialog {
+
+  constructor(oDialog, schemaService, oController) {
+    super(oDialog, schemaService, oController)
+    oController.byId("newSchemaAddButton").attachPress(this.submit, this)
+    oController.byId("newSchemaCancelButton").attachPress(this.cancel, this)
+    oController.byId("newSchemaName").attachChange(this.onNameChange, this)
+  }
+
+  resetValueState() {
+    this.oController.byId("newSchemaName").setValueState(sap.ui.core.ValueState.None);
+    this.oController.byId("newSchemaName").setValueStateText("");
+  }
+
+  isValid(oSchema) {
+    this.resetValueState();
+
+    let hasValidName = EntityValidationService.hasValidName(oSchema, "Schema",
+      this.oController.byId("newSchemaName"));
+
+    return hasValidName;
+  }
+
+  onNameChange() {
+    let sName = this.oDialog.getModel("entity").getProperty("/name");
+    if (GenericService.isValidEntityName(sName)) {
+      SchemaService.hasUniqueName(sName, this.oDialog.getModel("entity"))
+    } else {
+      this.oDialog.getModel("entity").setProperty("/newSchema/nameUnique", true)
+    }
+  }
+
+}
+
+class AddSchemaDialog extends SchemaDialog {
+
+  constructor(oDialog, schemaService, oController) {
+    super(oDialog, schemaService, oController);
+  }
+
+  onPress() {
+    this.oDialog.setModel(new sap.ui.model.json.JSONModel({
+      name: "",
+      description: "",
+      isEdit: false,
+      title: "Add"
+    }), "entity");
+
+    this.oDialog.open();
+  }
+
+}
+
+class EditSchemaDialog extends SchemaDialog {
+
+  constructor(oDialog, schemaService, oController) {
+    super(oDialog, schemaService, oController);
+  }
+
+  onPress() {
+    let current = this.oController._model.getProperty("/currentSchema");
+    current.isEdit = true;
+    current.title = "Edit";
+
+    this.oDialog.setModel(new sap.ui.model.json.JSONModel(jQuery.extend(true, {}, current)), "entity");
+    this.oDialog.open();
+  }
+
+}
+
+
+class MappingTableDialog extends EntityDialog {
+
+  constructor(oDialog, mappingTableService, schemaService, oController) {
+    super(oDialog, mappingTableService, oController)
+    this._schemaService = schemaService;
+    oController.byId("newMappingTableAddButton").attachPress(this.submit, this)
+    oController.byId("newMappingTableCancelButton").attachPress(this.cancel, this)
+    oController.byId("newMappingTableName").attachChange(this.onNameChange, this)
+    oController.byId("schemaNameSelect").attachChange(this.onSchemaSelect, this)
```

Comment:
Missing `;` 4x
Line `253` theoretically doesn't need it, because it is the last line of a function